File: frontend/src/types/shift.ts
export interface ShiftTemplate {
  id: string;
  tenant_id: string;
  name: string;
  name_ar: string | null;
  start_time: string;
  end_time: string;
  is_active: boolean;
  created_at: string;
}

export interface ShiftTemplateCreate {
  name: string;
  name_ar?: string;
  start_time: string;
  end_time: string;
}

export interface ShiftTemplateUpdate {
  name?: string;
  name_ar?: string;
  start_time?: string;
  end_time?: string;
  is_active?: boolean;
}

export interface Shift {
  id: string;
  tenant_id: string;
  driver_id: string;
  template_id: string | null;
  date: string;
  scheduled_start: string;
  scheduled_end: string;
  actual_start: string | null;
  actual_end: string | null;
  status: string;
  clock_in_method: string | null;
  clock_out_method: string | null;
  clock_in_selfie_url: string | null;
  clock_in_location_lat: number | null;
  clock_in_location_lng: number | null;
  clock_out_location_lat: number | null;
  clock_out_location_lng: number | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface ShiftCreate {
  driver_id: string;
  template_id?: string;
  date: string;
  scheduled_start: string;
  scheduled_end: string;
  notes?: string;
}

export interface ShiftFilters {
  date_from?: string;
  date_to?: string;
  driver_id?: string;
  status?: string;
  page?: number;
  per_page?: number;
}

export interface ShiftBulkAssign {
  driver_ids: string[];
  template_id: string;
  dates: string[];
}

export interface ShiftCalendarDay {
  date: string;
  shifts: (Shift & { driver_name?: string })[];
}

File: backend/app/schemas/shift.py
import uuid
from datetime import date, datetime, time

from pydantic import BaseModel


class ShiftTemplateCreate(BaseModel):
    name: str
    name_ar: str | None = None
    start_time: time
    end_time: time


class ShiftTemplateResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    name: str
    name_ar: str | None
    start_time: time
    end_time: time
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ShiftCreate(BaseModel):
    driver_id: uuid.UUID
    template_id: uuid.UUID | None = None
    date: date
    scheduled_start: datetime
    scheduled_end: datetime
    notes: str | None = None


class ShiftUpdate(BaseModel):
    template_id: uuid.UUID | None = None
    scheduled_start: datetime | None = None
    scheduled_end: datetime | None = None
    status: str | None = None
    notes: str | None = None


class ShiftResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    driver_id: uuid.UUID
    template_id: uuid.UUID | None
    date: date
    scheduled_start: datetime
    scheduled_end: datetime
    actual_start: datetime | None
    actual_end: datetime | None
    status: str
    clock_in_method: str | None
    clock_out_method: str | None
    clock_in_selfie_url: str | None
    clock_in_location_lat: float | None
    clock_in_location_lng: float | None
    clock_out_location_lat: float | None
    clock_out_location_lng: float | None
    notes: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ShiftTemplateUpdate(BaseModel):
    name: str | None = None
    name_ar: str | None = None
    start_time: time | None = None
    end_time: time | None = None
    is_active: bool | None = None

class ShiftBulkAssign(BaseModel):
    driver_ids: list[str]
    template_id: str | None = None
    dates: list[str]

class ShiftCalendarDay(BaseModel):
    date: str
    shifts: list[dict]
